File: src/fiefapp.entities/BuildingAlternative.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace fiefapp.entities
{
    public class BuildingAlternative
    {
        [BsonId]
        [BsonRepresentation(BsonType.ObjectId)]
        public string Id { get; set; }
        public string Type { get; set; }
        public decimal Upkeep { get; set; }
        public int Stonework { get; set; }
        public int Woodwork { get; set; }
        public int Smithswork { get; set; }
        public int Stone { get; set; }
        public int Wood { get; set; }
        public int Iron { get; set; }
    }
}

File: src/fiefapp.svelte/src/models/settings/building-alternative.ts
import { Building } from '../building';

interface IBuildingAlternative {
  type: string;
  upkeep: number;
  woodwork: number;
  stonework: number;
  smithswork: number;
  wood: number;
  stone: number;
  iron: number;
}

export class BuildingAlternative implements IBuildingAlternative {
  constructor(
    public type: string,
    public upkeep: number,
    public woodwork: number,
    public stonework: number,
    public smithswork: number,
    public wood: number,
    public stone: number,
    public iron: number
  ) {}

  public create(): Building {
    return new Building(
      this.type,
      this.upkeep,
      this.woodwork,
      this.stonework,
      this.smithswork,
      this.wood,
      this.stone,
      this.iron
    );
  }
}
